---
layout: page
title:  General installation for Fusion detection tutorial
summary: "Comand line preparation enviroment for our tutorial."
---
##  General installation for Fusion detection tutorial
## Links

* [Introduction  ]()
* [Software requirements]
* [Original Setup]()

  
___
##  Introduction


In this part of the course, we will present the main steps that are commonly used to process and to analyze RNA-seq cancer sequencing data for detect fusion transcript genes.
This workshop will show you how to launch individual steps of a complete RNA-seq pipeline for detect fusion with some software.
All the softwares are already installed on your virtual machine. If you need assistance please let  me know.

## Software requirements

Software we use need to be installed on recent linux machine. Exist however the possibility to use some virtual container like docker that allow to use this software also in another operative system (OS,Windows).
We concentrate our tutorial on some  software :

- Fusioncather [https://github.com/ndaniel/fusioncatcher]
- **STAR-FUSION** [https://github.com/STAR-Fusion/STAR-Fusion/wiki]
- **JAFFA** [https://github.com/Oshlack/JAFFA/wiki]
- PRADA [http://bioinformatics.mdanderson.org/main/PRADA:Overview]
- Defuse [https://bitbucket.org/dranew/defuse]


Some  of this tutorial for the dimension of the machine you have it is not possible to try.


## Original Setup

The initial structure of your folders should look like this:
```
$ tree tutorial-fusion-transcript-2017/
tutorial-fusion-transcript-2017/
├── RAW.gz
├── README.md
├── Results
│   ├── HCC1395
│   │   ├── chimerascan
│   │   │   └── chimeras.bedpe
│   │   ├── defuse
│   │   │   ├── gene_names.tsv
│   │   │   └── results.filtered.tsv
│   │   └── starfusion
│   │       └── starfusion.fusion_candidates.txt
│   ├── JAFFA
│   │   ├── BT474
│   │   ├── MCF7
│   │   └── Simulazione
│   └── STAR
│       ├── BT474
│       └── Simulazione
└── scripts
    ├── _run_starfusion.sh
    └── _script.jaffa.sh

13 directories, 8 files

```

If you don't have this directory please download this and  execute this commmand:




```{.bash
git clone https://bitbucket.org/biomauro/tutorial-fusion-transcript-2017
```
Here you have the basic structure of the data and the way to download all the data you need to start this tutorial.

Not all sofware was possibile to install in your virtual machines so we need to concentrate in two softwares.
In the next months  more information on the other programs are pubblish.

### Sample Information

For this tutorial we use 3 samples:

* BTF-474 (SRA accession SRR925695)[filter dataset](https://drive.google.com/open?id=0B1FwZagazjpcRnV4OFVUZlFIeUk)
* MCF-7 (SRA accession SRR925723)[filter dataset](https://drive.google.com/open?id=0B1FwZagazjpcRnV4OFVUZlFIeUk)

* testing Chimeric.out.junction of the BTF-474 from STAR-FUSION repository [link](https://github.com/STAR-Fusion/STAR-Fusion/tree/master/testing/__kickstart_data)

*  Fusion_simulation_50x (create in house)

*   Fusion of HCC1395 from  Andrew McPherson  tutorial 

The first two samples are used only for jaffa tutorial and are 2x76 bp

<img src="{{site.url}}/images/mpsQC__01sequenceQualityChart_BT4.png" width="600">
<img src="{{site.url}}/images/mpsQC__06sequenceLengthDistribution_BT4.png" width="600">

For the simulation we have this situation:

It is pair-end sequence 2x50 bp

```
$ java -jar /opt/bvatools-1.6/bvatools-1.6-full.jar readsqc --read1 ../RAW/reads_50x_R1.fq.gz  --read2 ../RAW/reads_50x_R2.fq.gz 

```


<img src="{{site.url}}/images/50x_q.png" width="600">


Al the tutorial  with the results exceed the space for singlo user. So wereduce the  files are in the  results.








