# Run the STAR RNA-Seq aligner on a sample

STAR-Fusion is a component of the Trinity Cancer Transcriptome Analysis Toolkit (CTAT). STAR-Fusion uses the STAR aligner to identify candidate fusion transcripts supported by Illumina reads. STAR-Fusion further processes the output generated by the STAR aligner to map junction reads and spanning reads to a reference annotation set[article]( http://biorxiv.org/content/early/2017/03/24/120295).
<a name='Installation'></a>
## Installation 

Exist many way to install STAR-Fusion. You can use conda recipes or you can install localy. 

```
 git clone --recursive https://github.com/STAR-Fusion/STAR-Fusion.git



```

### Tools Required:

*  STAR <https://github.com/alexdobin/STAR>
*  possibly some non-standard Perl modules - see below:
  
.

       A typical perl module installation may involve:
       perl -MCPAN -e shell
       install DB_File
       install URI::Escape
       install Set::IntervalTree
       install Carp::Assert
       install JSON::XS

<a name='ComputeRequirements'></a>
## Computing / Hardware requirements

If you're planning to run STAR to align reads to the human genome, then you'll need ~30G RAM.   If you've already run STAR and are just planning on running STAR-Fusion given the existing STAR outputs, then modest resources are required and it should run on any commodity hardware.



# Data Resources Required:

A reference genome and corresponding protein-coding gene annotation set, including blast-matching gene pairs must be provided to STAR-Fusion. We provide several alternative resources for human fusion transcript detection depending on whether you want to use GRCh37 or GRCh38 reference human genomes and corresponding Gencode annotation sets. Options are available here: https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/, so choose one, and below we refer to it as 'CTAT_resource_lib.tar.gz'. The gene annotations in each case are restricted to the protein-coding and lincRNA transcripts.

If you're looking to apply STAR-Fusion using a different target, you'll need to generate the required resources as described by our FusionFilter resource builder. FusionFilter comes included in the STAR-Fusion software.
<a name='Database'></a>
### Data Resources Required:

A reference genome and corresponding protein-coding gene annotation set, including blast-matching gene pairs must be provided to STAR-Fusion.  We provide several alternative resources for human fusion transcript detection depending on whether you want to use GRCh37 or GRCh38 reference human genomes and corresponding Gencode annotation sets.  Options are available here: <https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/>, so choose one, and below we refer to it as 'CTAT_resource_lib.tar.gz'.   The gene annotations in each case are restricted to the protein-coding and lincRNA transcripts.


If you're looking to apply STAR-Fusion using a different target, you'll need to generate the required resources as described by our [FusionFilter](http://FusionFilter.github.io) resource builder.  FusionFilter comes included in the STAR-Fusion software.


## Preparing the genome resource lib

If you downloaded the large (30G) 'plug-n-play' resource lib, then just untar/gz the archive and use it directly. 

####  Execution

Running STAR on paired end read fastqs can be performed in a single step.  Set the prefix argument `--outFileNamePrefix` to specify the location of the output.

Given paried-end of FASTQ files, run  STAR-Fusion like so:

``` {bash}

STAR-Fusion --genome_lib_dir /path/to/your/CTAT_resource_lib \
             --left_fq reads_1.fq \
             --right_fq reads_2.fq \
             --output_dir star_fusion_outdir
```
If you have single-end FASTQ files, just use the --left_fq parameter:

```{bash}
 STAR-Fusion --genome_lib_dir /path/to/your/CTAT_resource_lib \
             --left_fq reads_1.fq \ 
             --output_dir star_fusion_outdir
}



```
Also other programs are able to  detect fusion from  single-end. One of main problems are the lenght of the reads. It is important to have relatively long read (ex. at least 100 bp) otherwise you will be underpowerred for detecting fusion transcripts

## Kickstart mode: runninf STAR yourself, and  the running STAR--Fusion using eisting ouputs

It's not always the case that you want to have STAR-Fusion run STAR directly, as you may have already run STAR earlier on, or prefer to run STAR separately to use the outputs in other processes such as for expression estimates or variant detection.
Parameters that we recommend for running STAR as part of STAR-Fusion are as follows:

```{bash}
STAR --genomeDir ${star_index_dir} \                                             
      --readFilesIn ${left_fq_filename} ${right_fq_filename} \
      --twopassMode Basic \                                   
      --outReadsUnmapped None \                               
      --chimSegmentMin 12 \                                  
      --chimJunctionOverhangMin 12 \                         
      --alignSJDBoverhangMin 10 \                            
      --alignMatesGapMax 100000 \                            
      --alignIntronMax 100000 \                              
      --chimSegmentReadGapMax parameter 3 \                  
      --alignSJstitchMismatchNmax 5 -1 5 5 \
      --runThreadN ${THREAD_COUNT} \                           
      --limitBAMsortRAM 31532137230 \                        
      --outSAMtype BAM SortedByCoordinate 

```
This will (in part) generate a file called 'Chimeric.out.junction', which is used by STAR-Fusion like so:


```{bash}
STAR-Fusion --genome_lib_dir /path/to/your/CTAT_resource_lib \
             -J Chimeric.out.junction \
             --output_dir star_fusion_outdir --left_fq ${left_fq_filename} --right_fq ${right_fq_filename}

```

### Examples RUN

We need to detect fusion trascript of BT474 cells. Here we use only a small part of fastq. 

The virtual machine dimension not allow to run completely this tutorial. For this reason we use the kick start mode.


```{bash}
mkdir Tutorial_Fusion
mkdir example2
cd example2
STAR --genomeDir  /mnt/mp_working/DATABASE_TUTORIAL/GRCh37_gencode_v19_CTAT_lib_July192017/ctat_genome_lib_build_dir/ref_genome.fa.star.idx/ \
--readFilesIn /mnt/mp_working/Database_tutorial2017/tutorial-fusion-transcript-2017/RAW/BT474-demo_1.fastq.gz /mnt/mp_working/Database_tutorial2017/tutorial-fusion-transcript-2017/RAW/BT474-demo_2.fastq.gz  --twopassMode Basic --outReadsUnmapped None--chimSegmentMin 12 --chimJunctionOverhangMin 12 --alignSJDBoverhangMin 10 --alignMatesGapMax 100000 --alignIntronMax 100000  --chimSegmentReadGapMax parameter 3 --alignSJstitchMismatchNmax 5 -1 5 5 --runThreadN 3 --limitBAMsortRAM 31532137230 --outSAMtype BAM SortedByCoordinate --readFilesCommand zcat

```








```



### Run the star aligner

Running STAR on paired end read fastqs can be performed in a single step.  Set the prefix argument `--outFileNamePrefix` to specify the location of the output.

Assume we have end 1 and end 2 files as `$SAMPLE_FASTQ_1` and `$SAMPLE_FASTQ_2` environment variables pointing to paired fastq files for sample `$SAMPLE_ID`.

```{bash}
mkdir -p $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/
STAR --runThreadN 4 \
    --outSAMtype BAM SortedByCoordinate \
    --outSAMstrandField intronMotif \
    --outFilterIntronMotifs RemoveNoncanonicalUnannotated \
    --outReadsUnmapped None \
    --chimSegmentMin 15 \
    --chimJunctionOverhangMin 15 \
    --chimOutType WithinBAM \
    --alignMatesGapMax 200000 \
    --alignIntronMax 200000 \
    --genomeDir $STAR_GENOME_INDEX \
    --readFilesIn $SAMPLE_FASTQ_1 $SAMPLE_FASTQ_2 \
    --outFileNamePrefix $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/
```

> Note: In order to obtain alignments of chimeric reads potentially supporting fusions, we
> have added the `--chimSegmentMin 20` option to obtain chimerica reads anchored by at least
> 20nt on either side of the fusion boundary, and `--chimOutTypeWithinBAM` to report such
> alignments in the sam/bam output.

> Note: We are running with additional command line options for fusion discovery as described
> in the manual for STAR-Fusion.



The STAR fusion caller parses the chimeric alignments produced by the STAR aligner
and predicts fusions from these alignments.

```{bash}
STAR-Fusion \
    --chimeric_junction $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/Chimeric.out.junction \
    --genome_lib_dir $STAR_FUSION_GENOME_INDEX \
    --output_dir $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/starfusion
```

The fusion reads are output to the sam file `Chimeric.out.sam`.  To view these in IGV, first import into bam format, sort, and then index.

```{bash}
samtools view -bt $UCSC_GENOME_FILENAME \
    $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/Chimeric.out.sam \
    | samtools sort - -o $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/Chimeric.out.bam
samtools index $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/Chimeric.out.bam
igvtools count $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/Chimeric.out.bam \
    $TUTORIAL_HOME/analysis/star/$SAMPLE_ID/Chimeric.out.bam.tdf hg19
```

You can view the bam file of fusion reads in IGV at:

```
http://cbw#.dyndns.info/Module4/analysis/star/HCC1395/Chimeric.out.bam
```

where you need to replace # with your student ID.  Fusion reads can be found
at this location `chr2:160,832,184-160,837,535`.


